### PR TITLE
[GIE/Doc] Add initial design docs of GIE

### PR DESCRIPTION
**Pull request by longbinlai**
This PR tries to add the first design docs of GIE, including the design of IR layer, and the Codegen for different computing engines, etc.

### CODE REVIEW

File: docs/design_of_gie.md [hunk 1]
```diff
@@ -1,2 +1,143 @@
...
+core tasks in the graph query process, query verification and query optimization. Parsing for each
+language is completed in the corresponding client, and code generation for each execution engine is
+handled at the corresponding runtime. In this way, we only need to implement M parsers for M
+languages and N sets of code generation for N engines.
+
+:::{figure-md}
+
+<img src="images/design_of_gie.png"
+     alt="The architecture of GIE"
+     style="border-radius: 0.3125em; box-shadow: 0 2px 4px 0 rgba(34,36,38,.12),0 2px 10px 0 rgba(34,36,38,.08);"
```
Comment (yecol) [inline docs/design_of_gie.md:69]: Just a suggestion: to make it consistent and concise, maybe remove the style to make figures flat and borderless.

**Review by yecol — COMMENTED** [review]
Just a suggestion: to make it consistent and concise, maybe remove the style to make figures flat and borderless.

**Review by yecol — APPROVED** [review]

**Comment by sighingnow** [comment]
> Just a suggestion: to make it consistent and concise, maybe remove the style to make figures flat and borderless.

We could add a global style configuration for images, titles, etc. after finalize the documentation itself.

File: docs/design_of_gie.md (hunk 1 above)
Comment (longbinlai) [reply docs/design_of_gie.md:69]: Okay. I refine the other images as well.

**Pull request merged**